assets/src/apps/authoring/components/Modal/DiagnosticsWindow.tsx

```diff
@@ -63,13 +63,17 @@ const ActivityPartError: React.FC<{ error: any; onApplyFix: () => void }> = ({
         ? problem.item.rule.id
         : problem.type === DiagnosticTypes.INVALID_VALUE
         ? problem.item.rule.id
+        : problem.type === DiagnosticTypes.INVALID_EXPRESSION_VALUE
+        ? problem.item.rule.id
         : 'initState';
 
     const activity = result.meta.arg.activity;
-    const rule = activity.authoring.rules.find((rule: AdaptiveRule) => rule.id === ruleId);
+    if (activity) {
```

Comment:
is there a reason to expect that it might not be valid?

Reply:
Actually, till now we were checking the different validation in trap state so the ruleId was always present but with CAPI configData variables, part custom data validation, it's not present and was giving error during  `dispatch(setCurrentRule({ currentRule: rule })); ` hence added an layer for safety check.